test/constant.jl

```diff
@@ -5,6 +5,9 @@
 
     @test TimeDag.Constant(1.0) != TimeDag.Constant(1)
     @test !isequal(TimeDag.Constant(1.0), TimeDag.Constant(1))
+
+    @test TimeDag.Constant(1) == TimeDag.Constant{Int64}(1)
+    @test TimeDag.Constant{Number}(1) != TimeDag.Constant{Int64}(1)
```

Comment:
Why this choice of semantics, as opposed to aligning with `1 == 1.0`? I don't think I've seen a discussion of what definition of equality is useful for TimeDag come to think of it 🤷

Reply:
Deja vu...

Reply:
Weird. In "file" mode I can't see the other thread. But in "conversation" I can. 
![image](https://user-images.githubusercontent.com/7052222/183666112-646116b9-a925-416d-a604-76661a3a541d.png)

Reply:
I've added some additional documentation on this